File: lib/features/orders/domain/entities/order.dart
class OrderItem {
  final String id;
  final String offerId;
  final String offerTitle;
  final String? offerImageUrl;
  final double price;
  final int quantity;

  const OrderItem({
    required this.id,
    required this.offerId,
    required this.offerTitle,
    this.offerImageUrl,
    required this.price,
    required this.quantity,
  });

  double get total => price * quantity;
}

enum OrderStatus {
  pending,
  confirmed,
  preparing,
  ready,
  pickedUp,
  delivered,
  cancelled,
  rejected;

  String get displayName {
    switch (this) {
      case OrderStatus.pending:
        return 'Pending';
      case OrderStatus.confirmed:
        return 'Confirmed';
      case OrderStatus.preparing:
        return 'Preparing';
      case OrderStatus.ready:
        return 'Ready for Pickup';
      case OrderStatus.pickedUp:
        return 'Picked Up';
      case OrderStatus.delivered:
        return 'Delivered';
      case OrderStatus.cancelled:
        return 'Cancelled';
      case OrderStatus.rejected:
        return 'Rejected';
    }
  }
}

class Order {
  final String id;
  final String storeId;
  final String storeName;
  final String? storeLogoUrl;
  final List<OrderItem> items;
  final double subtotal;
  final double serviceFee;
  final double deliveryFee;
  final double discount;
  final double total;
  final OrderStatus status;
  final String? couponCode;
  final String paymentMethod;
  final bool isDelivery;
  final String? deliveryAddress;
  final DateTime createdAt;
  final DateTime? estimatedPickupTime;
  final DateTime? pickedUpAt;
  final String? notes;

  const Order({
    required this.id,
    required this.storeId,
    required this.storeName,
    this.storeLogoUrl,
    required this.items,
    required this.subtotal,
    this.serviceFee = 0,
    this.deliveryFee = 0,
    this.discount = 0,
    required this.total,
    required this.status,
    this.couponCode,
    required this.paymentMethod,
    this.isDelivery = false,
    this.deliveryAddress,
    required this.createdAt,
    this.estimatedPickupTime,
    this.pickedUpAt,
    this.notes,
  });
}

File: backend/src/LastHour.Domain/Entities/Order.cs
using LastHour.Domain.Common;
using LastHour.Domain.Exceptions;

namespace LastHour.Domain.Entities;

public class Order : BaseAuditableEntity
{
    public string CustomerId { get; set; } = string.Empty;
    public string StoreId { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";
    public decimal Subtotal { get; set; }
    public decimal ServiceFee { get; set; }
    public decimal DeliveryFee { get; set; }
    public decimal Discount { get; set; }
    public decimal Total { get; set; }
    public string? CouponCode { get; set; }
    public string PaymentMethod { get; set; } = "Cash";
    public bool IsDelivery { get; set; }
    public string? DeliveryAddress { get; set; }
    public double? DeliveryLatitude { get; set; }
    public double? DeliveryLongitude { get; set; }
    public DateTime? EstimatedPickupTime { get; set; }
    public DateTime? PickedUpAt { get; set; }
    public DateTime? DeliveredAt { get; set; }
    public string? Notes { get; set; }
    public string? CancellationReason { get; set; }

    public ICollection<OrderItem> Items { get; set; } = [];
    public ICollection<PaymentTransaction> PaymentTransactions { get; set; } = [];

    public void CalculateTotal()
    {
        Subtotal = Items.Sum(i => i.Price * i.Quantity);
        Total = Subtotal + ServiceFee + DeliveryFee - Discount;
        if (Total < 0) Total = 0;
    }

    public void Cancel(string? reason = null)
    {
        if (Status == "Delivered" || Status == "PickedUp")
            throw new BusinessRuleException("Cannot cancel a completed order");
        if (Status == "Cancelled")
            throw new BusinessRuleException("Order is already cancelled");
        Status = "Cancelled";
        CancellationReason = reason;
    }
}

public class OrderItem : BaseEntity
{
    public string OrderId { get; set; } = string.Empty;
    public string OfferId { get; set; } = string.Empty;
    public string OfferTitle { get; set; } = string.Empty;
    public string? OfferImageUrl { get; set; }
    public decimal Price { get; set; }
    public int Quantity { get; set; }

    public decimal Total => Price * Quantity;
}
